sklearn/tests/test_isotonic.py

```diff
@@ -42,9 +42,9 @@ def test_check_increasing_small_number_of_samples():
     x = [0, 1, 2]
     y = [1, 1.1, 1.05]
 
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
```

Comment:
Based on the source of `check_increasing`, the absence of warning we want to ensure is a UserWarning
```suggestion
    with warnings.catch_warnings():
        warnings.simplefilter("error", UserWarning)
```

Reply:
Same for following tests